### PR TITLE
Task/bws70 worker sign in

### PR DESCRIPTION
**Pull request by junojei**

### CODE REVIEW

File: shared/src/common/types/token/token-payload.type.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 type TokenPayload = {
   userId: string;
+  userRole: string;
```
Comment (what1s1ove) [inline shared/src/common/types/token/token-payload.type.ts]:
```suggestion
  userRole: UserRole;
```

File: backend/src/services/auth/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  ): Promise<null | EAMMasterSignInResponseDto | EAMWorkerSignInResponseDto> {
+    const { userId, userRole } = this.#tokenService.decode<TokenPayload>(token);
+
+    switch (userRole) {
+      case UserRole.master:
+        return this.#masterService.getCurrentUser(userId);
+      case UserRole.worker:
+        return this.#workerService.getCurrentUser(userId);
+      default:
+        return null;
```
Comment (what1s1ove) [inline backend/src/services/auth/auth.service.ts]:
```suggestion
      case UserRole.master: { // don't forget about curly
        return this.#masterService.getCurrentUser(userId);
       }
      case UserRole.worker: {
        return this.#workerService.getCurrentUser(userId);
        }
      default: {
        return null; // throw error here
      }
```

File: shared/src/common/enums/roles/user-role.enum.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+enum UserRole {
+  master = 'master',
+  worker = 'worker',
```
Comment (what1s1ove) [inline shared/src/common/enums/roles/user-role.enum.ts]:
```suggestion
  MASTER = 'master',
  WORKER = 'worker',
```

File: backend/src/services/master/master.service.ts [hunk 4]
```diff
@@ -70,14 +69,14 @@ class Master {
...
         userId: id,
+        userRole: UserRole.master,
       }),
     };
   }
 
   public async getCurrentUser(
-    token: string,
+    userId: string,
   ): Promise<EAMMasterSignUpResponseDto> {
```
Comment (Yaroslaww-1) [inline backend/src/services/master/master.service.ts]: This change will break backend\src\api\masters\masters.api.ts:53 - `.send(await masterService.getCurrentUser(token))`. So, you can either roll this changes back or remove /user route from master api. 
Also, `getCurrentUser` method which takes `userId` as a param bothers me. I'd rename it to `getUserById(userId: string)`

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

File: backend/src/services/master/master.service.ts (hunk 4 above)
Comment (junojei) [reply backend/src/services/master/master.service.ts]: I think so.

File: frontend/src/components/auth/auth.tsx [hunk 5]
```diff
@@ -1,17 +1,23 @@
-import { FC } from 'react';
+import { FC, useState } from 'react';
```
Comment (what1s1ove) [inline frontend/src/components/auth/auth.tsx]:
```suggestion
import { FC } from 'react';
```
import from hooks

**Review by Yaroslaww-1 — DISMISSED** [review]

File: frontend/src/components/common/header/header.tsx [hunk 6]
```diff
@@ -23,13 +27,16 @@ const Header: FC = () => {
...
+        <Link to={AppRoute.ROOT}>
           <img className={styles.logo} src={logo} alt="logo" />
         </Link>
         <div className={styles.menu}>
           <div className={styles.profile}>
             <span className={styles.profileName}>
-              {user?.email ?? '-'} @ {tenant?.name ?? '-'}
+              {(user as EAMMasterByIdResponseDto)?.email ??
+                (user as EAMWorkerByIdResponseDto)?.name ??
+                '-'}{' '}
```
Comment (Yaroslaww-1) [inline frontend/src/components/common/header/header.tsx]: It is quite hard to read. I think its better to move email and name to separate variables

File: frontend/src/components/auth/auth.tsx [hunk 7]
```diff
@@ -24,18 +35,47 @@ const Auth: FC = () => {
...
+    payload: EAMMasterSignInRequestDto,
+  ): void => {
+    dispatch(authActions.signInMaster(payload));
+  };
+
+  const handleWorkerSignInSubmit = (
+    payload: EAMWorkerSignInRequestDto,
+  ): void => {
+    dispatch(authActions.signInWorker(payload));
+    Object.keys(payload);
```
Comment (Yaroslaww-1) [inline frontend/src/components/auth/auth.tsx]: I think its not used anywhere

File: frontend/src/components/app/app.tsx [hunk 8]
```diff
@@ -36,7 +37,7 @@ const App: FC = () => {
     dispatch(appActions.getTenant({ id: user.tenantId }));
   }, [user, dispatch]);
 
-  if (!hasUser && hasToken) {
+  if (!hasUser && hasToken && authStatus !== DataStatus.REJECTED) {
```
Comment (Yaroslaww-1) [inline frontend/src/components/app/app.tsx:40]: What happens if auth would return error on every request? Isn't it a infinite loop?

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Review by Yaroslaww-1 — APPROVED** [review]

**Pull request merged**